src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts

```diff
@@ -109,7 +111,10 @@ export class TerminalProfileService implements ITerminalProfileService {
 		if (profilesChanged || contributedProfilesChanged) {
 			this._availableProfiles = profiles;
 			this._onDidChangeAvailableProfiles.fire(this._availableProfiles);
-			this._profilesReadyBarrier.open();
+			if (!this._profilesReadyBarrier.isOpen()) {
+				this._logService.debug('Took 20 seconds to get available terminal profiles');
```

Comment:
I think I got confused here, if it's not open it means it's less than 20 seconds. Opps, let's just revert this and do without the warning